lib/core/storage.js

```diff
@@ -20,32 +20,32 @@ export default class Storage {
       return this.removeUniversal(key)
     }
 
-    // Local state
-    this.setState(key, value)
-
     // Cookies
     this.setCookie(key, value)
 
     // Local Storage
     this.setLocalStorage(key, value)
 
+    // Local state
+    this.setState(key, value)
+
     return value
   }
 
   getUniversal (key) {
-    // Local state
```

Comment:
Would you please elaborate more on why we need to first read from cookies? (if the issue is `nuxt generate` we need to discuss more.)

Reply:
Talking with @JoaoPedroAS51, the reason was to keep tabs state in sync.